hand-eye-calibration/full_hand_eye_sample.py

```diff
@@ -0,0 +1,486 @@
+"""
+Script to set up communication, generate dataset and perform hand-eye calibration.
+"""
+from pathlib import Path
+import time
+import datetime
+
+import cv2
+import numpy as np
+import zivid
+import rtde.rtde as rtde
+import rtde.rtde_config as rtde_config
+
+
+def quat_to_rotm(quat: np.array) -> np.array:
+    """Convert from quaternion to rotation matrix
+
+    Args:
+        quat: Rotations in quaternions
+
+    Returns:
+        rot_matrix: Rotations as 3x3 rotation matrix
+    """
+
+    q_w = quat[0]
+    q_x = quat[1]
+    q_y = quat[2]
+    q_z = quat[3]
+
+    rot_matrix = np.zeros((3, 3))
+    rot_matrix[0, 0] = 1 - 2 * np.power(q_y, 2) - 2 * np.power(q_z, 2)
+    rot_matrix[0, 1] = 2 * q_x * q_y - 2 * q_z * q_w
+    rot_matrix[0, 2] = 2 * q_x * q_z + 2 * q_y * q_w
+    rot_matrix[1, 0] = 2 * q_x * q_y + 2 * q_z * q_w
+    rot_matrix[1, 1] = 1 - 2 * np.power(q_x, 2) - 2 * np.power(q_z, 2)
+    rot_matrix[1, 2] = 2 * q_y * q_z - 2 * q_x * q_w
+    rot_matrix[2, 0] = 2 * q_x * q_z - 2 * q_y * q_w
+    rot_matrix[2, 1] = 2 * q_y * q_z + 2 * q_x * q_w
+    rot_matrix[2, 2] = 1 - 2 * np.power(q_x, 2) - 2 * np.power(q_y, 2)
+
+    return rot_matrix
+
+
+def rotvec_to_quat(rotvec: np.array, angle: float = np.nan) -> np.array:
+    """Convert from rotation vector to quaternions
+
+    Args:
+        rotvec: Rotation vector
+        angle: Angle of rotation
+
+    Returns:
+        quat: Rotation in quaternions
```

Comment:
There's no need to name the return-values in the docstring. These names aren't visible from the callers perspective anyway.

Reply:
Done